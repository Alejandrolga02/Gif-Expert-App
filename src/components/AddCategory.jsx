import PropTypes from "prop-types";
import { useState } from "react";

export const AddCategory = ({ onNewCategory }) => {
	const [inputValue, setInputValue] = useState("");

	const onInputChange = ({ target }) => setInputValue(target.value);

	const onSubmit = (event) => {
		event.preventDefault();

		const cleanCategory = inputValue.trim();

		if (cleanCategory.trim().length <= 2) return;

		onNewCategory(cleanCategory);
		setInputValue("");
	};

	return (
		<form onSubmit={onSubmit}>
			<input type="text" placeholder="Search Gifs" value={inputValue} onChange={onInputChange} />
			<button type="submit" className="btn-search">
				Search GIFs
			</button>
		</form>
	);
};

AddCategory.propTypes = {
	onNewCategory: PropTypes.func,
};
