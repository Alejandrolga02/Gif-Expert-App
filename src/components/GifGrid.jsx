import PropTypes from "prop-types";

import { GifItem } from "./";
import { useFetchGifs } from "../hooks";

export const GifGrid = ({ category, deleteCategory }) => {
	const { images, isLoading } = useFetchGifs(category);

	const onDeleteCategory = () => {
		deleteCategory(category);
	};

	return (
		<>
			<div className="heading-container">
				<h3>{category}</h3>
				<button aria-label="btn-delete" onClick={onDeleteCategory} className="btn-delete">
					Delete
				</button>
			</div>
			<div className="card-grid">
				{isLoading && <h2>Cargando...</h2>}
				{images.map((image) => (
					<GifItem key={image.id} {...image} />
				))}
			</div>
		</>
	);
};

GifGrid.propTypes = {
	category: PropTypes.string.isRequired,
	deleteCategory: PropTypes.func.isRequired,
};
