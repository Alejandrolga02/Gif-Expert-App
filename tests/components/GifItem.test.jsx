import { render, screen } from "@testing-library/react";
import { GifItem } from "../../src/components/GifItem";

describe("tests in GifItem", () => {
	const id = "asdfasdf";
	const title = "One Punch";
	const url = "https://upload.wikimedia.org/wikipedia/commons/0/09/One_Punch_Man_logo.svg";

	test("should make match with snapshot", () => {
		const { container } = render(<GifItem title={title} id={id} url={url} />);

		expect(container).toMatchSnapshot();
	});

	test("should show an image with the url and alt apropiated", () => {
		// expect(screen.getByRole("img").src).toBe(url);
		render(<GifItem title={title} id={id} url={url} />);
		const { src, alt } = screen.getByRole("img");
		expect(src).toBe(url);
		expect(alt).toBe(title);
	});

	test("should show title in the component", () => {
		render(<GifItem title={title} id={id} url={url} />);

		expect(screen.getByText(title)).toBeTruthy();
	});
});
