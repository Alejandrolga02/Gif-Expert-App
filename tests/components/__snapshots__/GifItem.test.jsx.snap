// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`tests in GifItem should make match with snapshot 1`] = `
<div>
  <div
    class="card"
  >
    <img
      alt="One Punch"
      src="https://upload.wikimedia.org/wikipedia/commons/0/09/One_Punch_Man_logo.svg"
    />
    <p>
      One Punch
    </p>
  </div>
</div>
`;
